robot/board.py

```diff
@@ -50,23 +55,34 @@ def _send(self, message, retry=False):
                     raise ConnectionError(self._get_lc_error())
                 self._send(message, retry=True)  # Retry Recursively
 
-    def _recv(self, retry=False):
-
+    def _recv(self, _retry=False):
         """
         Receive a message from the robotd socket
-        :return: message
+        :return: the message
         """
-        # TODO split receieves by \n characters and return them one at a time.
-        # Currently this is mitigated by having a large receive buffer, but I'd rather
-        # we did it properly at some point
-        try:
-            return self.sock.recv(Board.RECV_BUFFER)
-        except (socket.timeout, BrokenPipeError, OSError):
-            if retry:
-                raise ConnectionError(self._get_lc_error())
-            else:
-                self._connect(self.sock_path)  # Reconnect
-                return self._recv(retry=True)  # Retry recursively
+
+        # TODO catch empty string receives as bad things
+        unended_text = b''
+
+        while not self._recv_buffer:
```

Comment:
I like it, implemented something similar but even more human readable, see 6a3bfe7